extensions/spark/kyuubi-spark-authz/pom.xml

```diff
@@ -323,6 +324,13 @@
             <artifactId>scala-collection-compat_${scala.binary.version}</artifactId>
             <scope>test</scope>
         </dependency>
+
+        <dependency>
+            <groupId>org.apache.hudi</groupId>
+            <artifactId>hudi-spark3.4-bundle_2.12</artifactId>
+            <version>${hudi.version}</version>
+            <scope>provided</scope>
+        </dependency>
```

Comment:
1. Move the dependency definition to the project parent pom.
2. use place holder for scala version and hudi vrsion
3. set scope to `test`

Check the counter part in pom for Iceberg for reference, which is very helpful in your case.

Reply:
1. First, if I add this jar to parent pom, will have a code conflict , since hudi bundle jar miss shade some class, cause conflict then I move to this place
2. Hudi only support scala 2.12 now
3. Done

Reply:
The conflict is 
```
Error: ] /home/runner/work/kyuubi/kyuubi/kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/PrometheusReporterService.scala:48: type mismatch;
 found   : MetricRegistry (in com.codahale.metrics) 
 required: MetricRegistry (in org.apache.hudi.com.codahale.metrics) 
Error: [ERROR] one error found
```